nflow-engine/src/main/java/io/nflow/engine/internal/executor/WorkflowStateProcessor.java

```diff
@@ -204,9 +205,14 @@ private WorkflowInstance saveWorkflowInstanceState(StateExecutionImpl execution,
       logger.debug("No handler method defined for {}, clearing next activation", execution.getNextState());
       execution.setNextActivation(null);
     }
+    WorkflowState nextState = definition.getState(execution.getNextState());
+    if (definition.getSettings().wakeupParentWhenFinished && nextState.getType() == WorkflowStateType.end
+        && instance.parentWorkflowId != null) {
```

Comment:
It would be faster to reverse the order of checks here. Since looking at settings is slower than looking at instance/nextState varibles (but the difference is not measurable so I do not know why I wrote this comment)

Reply:
Fixed anyway